src/main/java/org/apache/sling/models/impl/injectors/OSGiServiceInjector.java

```diff
@@ -79,9 +82,24 @@ public Object getValue(@NotNull Object adaptable, String name, @NotNull Type typ
                 filterString = filter.value();
             }
         }
+        if (adaptable instanceof SlingHttpServletRequest && StringUtils.isBlank(filterString)) {
+            SlingHttpServletRequest request = (SlingHttpServletRequest) adaptable;
+            Object service = getValueFromServiceCache(request, type.getClass());
+            if (service != null) {
+                return service;
+            }
+        }
         return getValue(adaptable, type, filterString, callbackRegistry);
     }
 
+    private <T> Object getValueFromServiceCache(final SlingHttpServletRequest request, Class<T> type) {
```

Comment:
SlingBindings is not really a service cache, therefore I would rather call this method getServiceFromBindings()